components/test/src/main/java/org/trellisldp/test/SimpleResourceTemplate.java

```diff
@@ -0,0 +1,128 @@
+import static java.util.Optional.of;
+import static java.util.Optional.ofNullable;
+
+import java.util.AbstractMap.SimpleImmutableEntry;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.apache.commons.rdf.api.Dataset;
+import org.apache.commons.rdf.api.IRI;
+import org.apache.commons.rdf.api.Quad;
+import org.trellisldp.api.BinaryTemplate;
+import org.trellisldp.api.ResourceTemplate;
+import org.trellisldp.vocabulary.LDP;
+import org.trellisldp.vocabulary.OA;
+import org.trellisldp.vocabulary.Trellis;
+
+/**
+ * A template for generating a {@link org.trellisldp.api.Resource} in a persistence layer.
+ */
+public class SimpleResourceTemplate implements ResourceTemplate {
+
+    private final IRI identifier;
+    private final IRI interactionModel;
+    private final IRI container;
+    private final Dataset dataset;
+    private final BinaryTemplate binary;
+
+    /**
+     * Create a template for generating a Trellis Resource.
+     * @param identifier the identifier
+     * @param interactionModel the LDP interaction model
+     * @param dataset the dataset
+     * @param container the container, may be {@code null}
+     * @param binary the binary template, may be {@code null}
+     */
+    public SimpleResourceTemplate(final IRI identifier, final IRI interactionModel, final Dataset dataset,
+            final IRI container, final BinaryTemplate binary) {
+        requireNonNull(identifier, "identifier may not be null!");
+        requireNonNull(interactionModel, "interactionModel may not be null!");
+        this.identifier = identifier;
+        this.interactionModel = interactionModel;
+        this.container = container;
+        this.dataset = dataset;
+        this.binary = binary;
+    }
+
+    @Override
+    public IRI getIdentifier() {
+        return identifier;
+    }
+
+    @Override
+    public IRI getInteractionModel() {
+        return interactionModel;
+    }
+
+    @Override
+    public Optional<IRI> getMembershipResource() {
+        return firstValue(LDP.membershipResource);
+    }
+
+    @Override
+    public Optional<IRI> getMemberOfRelation() {
+        return firstValue(LDP.isMemberOfRelation);
+    }
+
+    @Override
+    public Optional<IRI> getMemberRelation() {
+        return firstValue(LDP.hasMemberRelation);
+    }
+
+    @Override
+    public Optional<IRI> getInsertedContentRelation() {
+        return firstValue(LDP.insertedContentRelation);
+    }
+
+    @Override
+    public Optional<IRI> getContainer() {
+        return ofNullable(container);
+    }
+
+    @Override
+    public Optional<BinaryTemplate> getBinaryTemplate() {
+        return ofNullable(binary);
+    }
+
+    @Override
+    public Dataset dataset() {
+        return dataset;
+    }
+
+    @Override
+    public Stream<? extends Quad> stream() {
+        return dataset.stream();
+    }
+
+    @Override
+    public Stream<Entry<String, String>> getExtraLinkRelations() {
```

Comment:
If we're going to leave a computation here, we ought to explain in a Javadoc why  `OA.annotationService`and `LDP.inbox` merit this treatment, no?